File: ZentavioCRM/StaffingManagementSystem.Web/src/services/auditLogService.ts
/**
 * Audit log API — thin wrapper around ZentavioCRM.Api's AuditLogsController.
 */
import { apiClient } from "@/services/apiClient";
import { callApi } from "@/services/apiHelpers";

export interface AuditLogEntry {
  id: string;
  entityType: string;
  entityId: string;
  action: string;
  summary: string;
  performedByUserName: string | null;
  createdAtUtc: string;
}

const getForEntity = (entityType: string, entityId: string) =>
  callApi<AuditLogEntry[]>(apiClient.get("/api/audit-logs", { params: { entityType, entityId } }));

export const auditLogService = { getForEntity };

File: ZentavioCRM/StaffingManagementSystem.Services/AuditLogService.cs
using ZentavioCRM.Core.DTOs.Audit;
using ZentavioCRM.Core.Entities;
using ZentavioCRM.Repositories.Interfaces;
using ZentavioCRM.Services.Interfaces;

namespace ZentavioCRM.Services
{
    /// <inheritdoc cref="IAuditLogService"/>
    public class AuditLogService : IAuditLogService
    {
        private readonly IAuditLogRepository _auditLogRepository;

        public AuditLogService(IAuditLogRepository auditLogRepository)
        {
            _auditLogRepository = auditLogRepository;
        }

        public Task LogAsync(string entityType, Guid entityId, string action, string summary, Guid? performedByUserId)
            => _auditLogRepository.AddAsync(new AuditLog
            {
                EntityType = entityType,
                EntityId = entityId,
                Action = action,
                Summary = summary,
                PerformedByUserId = performedByUserId,
                CreatedAtUtc = DateTime.UtcNow,
            });

        public async Task<IReadOnlyList<AuditLogDto>> GetForEntityAsync(string entityType, Guid entityId)
        {
            var logs = await _auditLogRepository.GetForEntityAsync(entityType, entityId);

            return logs.Select(l => new AuditLogDto
            {
                Id = l.Id,
                EntityType = l.EntityType,
                EntityId = l.EntityId,
                Action = l.Action,
                Summary = l.Summary,
                PerformedByUserName = l.PerformedByUser?.FullName,
                CreatedAtUtc = l.CreatedAtUtc,
            }).ToList();
        }
    }
}
